esp-hal-common/src/peripheral.rs

```diff
@@ -282,6 +282,18 @@ mod peripheral_macros {
             $(
                 pub use peripherals::$name;
             )*
+
+            $(
+                $(
+                    $(
+                        paste::paste!{
+                            pub fn [<bind_ $interrupt:lower _interrupt >](_peripheral: &mut peripherals::$name, handler: unsafe extern "C" fn() -> ()) {
```

Comment:
I struggle to parse macros, so I might be missing something here, should this just be a method on the peripheral itself?

E.g

```rust
impl GPIO {
 fn bind_interrupt(&mut self, handler: extern "C" fn()) {}
}
```

Reply:
That's brilliant! Much better

Reply:
We should also cfg this method away if neither of the rt features are enabled too, that would be a fun one to debug if we didn't :D.